### DISCUSSION TITLE
Help Needed: Vercel Deployment Issue with Serverless Function Accessing File Folder

### DISCUSSION THREAD

**Discussion by Typogram** [vercel/community#4949] [Help]
### Summary

Hi everyone,

I'm facing a deployment issue with my SvelteKit-based serverless function on Vercel, and I'd appreciate any guidance or advice from this knowledgeable community.

**Problem Overview:**
- **Functionality:** My serverless function is designed to read from a folder that contains handlebar templates. It then compiles these templates into files and uploads them to GitHub using the GitHub API.
- **Local Environment:** Everything works perfectly on my local machine.
- **Deployment Issue:** However, when deployed to Vercel, the function fails to locate the folder.

**Specifics:**
- The folder in question, `repo-template`, is supposed to be in the root directory. Here's a screenshot showing the output folder structure on Vercel: 
![Screenshot 2023-11-23 151814](https://github.com/vercel/community/assets/41246474/9f4f6c0c-77bf-4a31-a7ff-1a65485bf009)

- My current code attempts to locate the folder using this path:
  ```javascript
  const templatesDir = path.join(process.cwd(), 'repo-template');
  ```
  This translates to `/var/task/repo-template` on the server, which seems to be incorrect.

**Seeking Help:**
- How should I modify my code to correctly point to the `repo-template` folder that is deployed on the server?
- Is there a different approach or a best practice for handling file system access in a serverless environment on Vercel, particularly with SvelteKit?

Any insights, code snippets, or links to resources would be greatly appreciated. I'm really stuck on this, and your expert advice could be a game-changer!

Thanks in advance for your help!

### Steps to Reproduce (My code for +server.ts

```
async function generateFilesFromTemplates(templateData) {
  const templatesDir = path.join(
    process.cwd(),
    'repo-template'
  );

  console.log('templatesDir', templatesDir);
  const files = [];

  // Recursively read through all files in the templates directory
  const readTemplates = async (dir) => {
    const directoryEntries = await fs.readdir(dir, { withFileTypes: true });

    for (const dirent of directoryEntries) {
      const fullPath = path.join(dir, dirent.name);

      if (dirent.isDirectory()) {
        await readTemplates(fullPath);
      } else if (dirent.isFile() && path.extname(dirent.name) === '.hbs') {
        const templateContent = await fs.readFile(fullPath, 'utf-8');
        const template = Handlebars.compile(templateContent);
        const outputContent = template(templateData);
        const outputPath = fullPath
          .replace(templatesDir, '')
          .replace(/\.hbs$/, '');

        // Add file to the array, removing '.hbs' from the filename
        files.push({
          path: outputPath,
          content: outputContent
        });
      }
    }
  };

  await readTemplates(templatesDir);

  return files;
}
```

**Comment by Typogram** [comment]
Also tried
```
  const templatesDir = path.join(process.cwd(), '../../repo-template');
```
which will point to

```
/repo-template
```
still not correct, getting the same error:
```
[Error: ENOENT: no such file or directory, scandir '/repo-template'] { errno: -2, code: 'ENOENT', syscall: 'scandir', path: '/repo-template' }
```

**Comment by Yaucrates** [comment]
Have this issue too, did you ever find a fix?
